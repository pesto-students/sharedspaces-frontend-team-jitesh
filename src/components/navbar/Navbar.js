import React from 'react'
import { useSelector } from 'react-redux'
import Button from '../button/Button'
import { Link } from 'react-router-dom'
import './navbar.scss'
import AdminNavbar from '../adminNavbar/AdminNavbar';
import ProfileDropdown from '../profileDropdown/ProfileDropdown'

export default function Navbar(props) {
    const userDetail = useSelector(state => state.site.userDetail)

    const { location } = props;
    if (location?.pathname?.match(/login/) ||
        location?.pathname?.match(/sign-up/)
    ) {
        document.querySelector('body').style.margin = 0
        return null;
    } else if (location?.pathname?.match(/admin/)) {
        return <AdminNavbar />
    }

    document.querySelector('body').style.marginTop = "80px"
    return (
        <nav className="navbar text-gray-800 fixed top-0 left-0 right-0 w-full flex flex-wrap items-center justify-between font-bold shadow-new navbar navbar-expand-lg navbar-light">
            <div className="left-section flex items-center">
                <Link to="/">
                    <img className='logo' src="/assets/images/logo-black.png" alt="" />
                </Link>

                <nav className="flex sm:justify-center space-x-4 ml-10">
                    {[
                        ['Explore', '/search'],
                        // ['Liked Locations', '/liked-locations'],
                    ].map(([title, url]) => (
                        <Link to={url} className="rounded px-3 py-2 text-slate-700 font-medium hover:bg-slate-100 hover:text-slate-900">{title}</Link>
                    ))}
                </nav>
            </div>
            {userDetail === null
                ? <div className="right-section flex items-center">
                    <Link to="/login">
                        <Button buttonType="primary" className="mx-2">Login</Button>
                    </Link>
                    <Link to="/sign-up">
                        <Button buttonType="primary-outline" className="ml-2">Sign Up</Button>
                    </Link>
                </div>
                : <div className="right-section flex items-center">
                    <ProfileDropdown userDetail={userDetail} />
                </div>}

        </nav>
    )
}
