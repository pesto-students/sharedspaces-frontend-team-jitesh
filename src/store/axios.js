import axios from "axios";

const Axios = axios.create({
  baseURL: "http://localhost:5000/api",
  // baseURL:
  //   process.env.REACT_APP_BASE_URL,
});

export default Axios;
