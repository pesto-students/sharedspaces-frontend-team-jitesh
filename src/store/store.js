import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";
import { combineReducers } from "redux";
import siteReducer from "./reducers/siteReducer";

let middleware = [thunk];

const rootReducer = combineReducers({
  site: siteReducer,
});

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(...middleware))
);

export default store;
