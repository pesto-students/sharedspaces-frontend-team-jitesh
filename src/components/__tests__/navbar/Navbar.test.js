import { render, screen, cleanup } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from 'react-router-dom';
import store from '../../../store/store'
import renderer from 'react-test-renderer'
import Navbar from '../../navbar/Navbar'

afterEach(() => {
    cleanup()
})


test('test matches snapshot', () => {
    const component = renderer.create(
        <BrowserRouter>
            <Provider store={store}>
                <Navbar />
            </Provider>
        </BrowserRouter>
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})
