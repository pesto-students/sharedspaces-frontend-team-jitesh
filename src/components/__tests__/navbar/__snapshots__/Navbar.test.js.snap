// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<nav
  className="navbar text-gray-800 fixed top-0 left-0 right-0 w-full flex flex-wrap items-center justify-between font-bold shadow-new navbar navbar-expand-lg navbar-light"
  data-testid="navbar"
>
  <div
    className="left-section flex items-center"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="logo"
        src="/assets/images/logo-black.png"
      />
    </a>
    <nav
      className="menu-list flex sm:justify-center space-x-4 ml-10"
    >
      <a
        className="rounded px-3 py-2 text-slate-700 font-medium hover:bg-slate-100 hover:text-slate-900"
        href="/search"
        onClick={[Function]}
      >
        Explore
      </a>
    </nav>
  </div>
  <div
    className="right-section flex items-center"
  >
    <button
      className="
            btn-wrapper
        bg-red-500 text-white border-2 border-red-500 hover:bg-red-500 hover:shadow-sm active:bg-red-800 active:shadow-sm active:text-white
        false
        false
        false
        undefined
        px-6 py-1.5 font-medium rounded shadow-md transition duration-150 ease-in-out  whitespace-nowrap cursor-pointer"
      data-testid="button"
      onClick={[Function]}
    >
      List Your Space
    </button>
    <span
      className="text-gray-200 font-light text-4xl mx-4"
    >
      |
    </span>
    <div>
      <a
        href="/login"
        onClick={[Function]}
      >
        <button
          className="
            btn-wrapper
        bg-red-500 text-white border-2 border-red-500 hover:bg-red-500 hover:shadow-sm active:bg-red-800 active:shadow-sm active:text-white
        false
        false
        false
        mr-2
        px-6 py-1.5 font-medium rounded shadow-md transition duration-150 ease-in-out  whitespace-nowrap cursor-pointer"
          data-testid="button"
        >
          Login
        </button>
      </a>
      <a
        href="/sign-up"
        onClick={[Function]}
      >
        <button
          className="
            btn-wrapper
        false
        bg-white text-red-500 border-red-500 border-2 hover:bg-red-500 hover:text-white hover:shadow-sm active:bg-red-800 active:shadow-sm active:text-white
        false
        false
        ml-2
        px-6 py-1.5 font-medium rounded shadow-md transition duration-150 ease-in-out  whitespace-nowrap cursor-pointer"
          data-testid="button"
        >
          Sign Up
        </button>
      </a>
    </div>
  </div>
  <div
    className="mobile-dropdown"
  >
    <button
      className="dropdown-wrapper text-3xl text-red-500 flex items-center h-15 px-2 py-1 transition duration-150 ease-in-out  whitespace-nowrap cursor-pointer"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
    >
      <i
        class="fa-solid fa-bars"
      />
    </button>
  </div>
</nav>
`;
